### PR TITLE
build: build seastar in build directory

### PR DESCRIPTION
**Pull request by avikivity**
Currently, seastar is built in seastar/build/{mode}. This means we have two build
directories: build/{mode} and seastar/build/{mode}.

This patch changes that to have only a single build directory (build/{mode}). It
does that by calling Seastar's cmake directly instead of through Seastar's
./configure.py.  However, to support dpdk, if that is enabled it calls cmake
through Seastar's ./cooking.sh (similar to what Seastar's ./configure.py does).

All ./configure.py flags are translated to cmake variables, in the same way that
Seastar does.

Tests: unit (dev), build with and without dpdk, peek at compiler flags while the
       compiler is running.

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -1438,7 +1453,7 @@ def configure_zstd(build_dir, mode):
         rule configure
           command = {python} configure.py $configure_args
           generator = 1
-        build build.ninja: configure | configure.py seastar/configure.py
+        build build.ninja: configure | configure.py seastar/CMakeLists.txt
```
Comment (denesb) [inline configure.py]: I think the `seastar/*` dependency is unnecessary in this rule. We already unconditionally invoke `ninja .../libseastar.a` regardless of the target built. CMake will detect if any of its `CMakeLists.txt` files is dirty and will re-run itself.

Comment (avikivity) [reply configure.py]: ok, will drop (it's very partial in any case)

**Comment by avikivity** [comment]
updated: dropped dependency on seastar/CMakeLists.txt according to @denesb recommendation.

File: configure.py (hunk 1 above)
Comment (denesb) [reply configure.py]: Yes, there is a CMakeLists.txt in every directory, not just the top-level one.

**Comment by avikivity** [comment]
@scylladb/scylla-maint ping

**Comment by avikivity** [comment]
Dequeued since it fails next checks. I see @espindola already sent a fix, I'll merge it into the original patch and resubmit.

**Pull request merged**